ietf/api/tests.py

```diff
@@ -222,6 +222,70 @@ def test_api_set_session_video_url(self):
         event = doc.latest_event()
         self.assertEqual(event.by, recman)
 
+    def test_api_set_meetecho_recording_name(self):
+        url = urlreverse("ietf.meeting.views.api_set_meetecho_recording_name")
+        recmanrole = RoleFactory(group__type_id="ietf", name_id="recman")
+        recman = recmanrole.person
+        meeting = MeetingFactory(type_id="ietf")
+        session = SessionFactory(group__type_id="wg", meeting=meeting)
+        apikey = PersonalApiKey.objects.create(endpoint=url, person=recman)
+        name = "testname"
+
+        # error cases
+        r = self.client.post(url, {})
+        self.assertContains(r, "Missing apikey parameter", status_code=400)
+
+        badrole = RoleFactory(group__type_id="ietf", name_id="ad")
+        badapikey = PersonalApiKey.objects.create(endpoint=url, person=badrole.person)
+        badrole.person.user.last_login = timezone.now()
+        badrole.person.user.save()
+        r = self.client.post(url, {"apikey": badapikey.hash()})
+        self.assertContains(r, "Restricted to role: Recording Manager", status_code=403)
+
+        r = self.client.post(url, {"apikey": apikey.hash()})
+        self.assertContains(r, "Too long since last regular login", status_code=400)
```

Comment:
If we don't already, we ought to cover features like this in a unit test of the `require_api_key()` decorator and trust that. (Not actually asking for a change here unless you're enthusiastic about it, just pointing it out.)

Reply:
I felt the lack-of-reuse pain, but didn't pursue it as I think we might want to rework _all_ of these to use DRF going forward maybe?